bridge-token-factory/src/lib.rs

```diff
@@ -224,6 +291,44 @@ impl BridgeTokenFactory {
         self.tokens.iter().collect::<Vec<_>>()
     }
 
+     /// Finish updating token metadata once the proof was successfully validated. 
+     /// Can only be called by the contract itself.
+    pub fn finish_updating_metadata(
+        &mut self,
+        #[callback]
+        #[serializer(borsh)]
+        verification_success: bool,
+        #[serializer(borsh)] token: String,
+        #[serializer(borsh)] name: String,
+        #[serializer(borsh)] symbol: String,
+        #[serializer(borsh)] decimals: u8,
+        #[serializer(borsh)] timestamp: u64,
+        #[serializer(borsh)] proof: Proof,
+    ) {
+        assert_self();
+        assert!(verification_success, "Failed to verify the proof");
+        let required_deposit = self.record_proof(&proof);
+
+        assert!(
+            env::attached_deposit()
+                >= required_deposit + self.bridge_token_storage_deposit_required
+        );
+        env::log(format!("Finish updating metadata. Name:{} Symbol:{:?} Decimals:{:?} at : {:?}", name, symbol, decimals, timestamp).as_bytes());
+        let reference: Option<String> = Some(String::default());
+        let reference_hash: Option<Base64VecU8> = Some(Base64VecU8(vec![]));
+        // if timestamp > last updated at 
+        ext_bridge_token::set_metadata(
+            name.into(),
+            symbol.into(),
+            reference, 
+            reference_hash,
+            decimals.into(),
```

Comment:
Why do you overwrite token's `reference` and `reference_hash` with empty data? As these fields don't exist in the `MetadataConnector`'s `Log`, just set those to `None` so they won't be updated in `set_metadata()` method.